doc/neps/nep-0030-duck-array-protocol.rst

```diff
@@ -77,6 +90,47 @@ libraries that don't already implement ``__array__`` but would like to use duck
 array typing, it is advised that they they introduce both ``__array__`` and
 ``__duckarray__`` methods.
 
+Usage
+-----
+
+An example of how the ``__duckarray__`` protocol could be used to write a
+``concatenate`` function, and its produced outcome, can be seen below. Note that
+here we are assuming Dask has implemented the ``__duckarray__`` method.
+
+.. code:: python
+
+    def duckarray_concatenate(a, b):
```

Comment:
I think a duck array `stack` written in terms of `np.concatenate` would be a better example here, because it shows the purpose of `np.duckarray` -- for guaranteeing that attributes like `shape` and `ndim` exist.

You could probably just lightly adapt the actual implementation of `np.stack`, perhaps removing some error checking:
https://github.com/numpy/numpy/blob/v1.17.0/numpy/core/shape_base.py#L419

Reply:
Agreed, please let me know what you think of the new example, I've capped it significantly to make it as simple as possible.